From 55561144a824fbd85325a1aef54bef1926690b07 Mon Sep 17 00:00:00 2001
From: Guoqing Jiang <Guoqing.Jiang@windriver.com>
Date: Wed, 11 Sep 2013 14:20:24 +0800
Subject: [PATCH 47/57] ARM/keystone: use use CLKSRC_OF init to init arm arch
 timer

Commit 0583fe478a7d93be2c814b7e50d6e81c287edfe8 "ARM: convert
arm/arm64 arch timer to use CLKSRC_OF init" converts arm and
arm64 to use CLKSRC_OF DT based initialization for the arch
timer, make keystone also adapt to this change.

Signed-off-by: Guoqing Jiang <Guoqing.Jiang@windriver.com>
---
 arch/arm/mach-keystone/keystone.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/arch/arm/mach-keystone/keystone.c b/arch/arm/mach-keystone/keystone.c
index e688b50..b6959b5 100644
--- a/arch/arm/mach-keystone/keystone.c
+++ b/arch/arm/mach-keystone/keystone.c
@@ -13,6 +13,7 @@
  * You should have received a copy of the GNU General Public License along with
  * this program.  If not, see <http://www.gnu.org/licenses/>.
  */
+#include <linux/clocksource.h>
 #include <linux/delay.h>
 #include <linux/dma-mapping.h>
 #include <linux/io.h>
@@ -252,13 +253,7 @@ static void __init keystone_timer_init(void)
 	if (!error)
 		return;
 
-	error = arch_timer_of_register();
-	if (!error) {
-		arch_timer_sched_clock_init();
-		return;
-	}
-
-	panic("no timer!\n");
+	clocksource_of_init();
 }
 
 static bool is_coherent(struct device *dev)
-- 
1.8.4.93.g57e4c17

